git-sync/atomic.md

```diff
@@ -0,0 +1,73 @@
+# Atomic initial clone
+
+To encourage sidecar functionality, git-sync now provides the option to make your initial clone atomic. Many use cases would require that the target repo directory be completely populated before another container begins accessing the content. 
+
+For example, a webserver container might run alongside the git-sync sidecar and serve content from the Git repo. The webserver would not want to begin serving the content in its target directory until the data is fully pulled into the volume by git-sync.  The process involves cloning and pulling initially to a temporary directory, setting a symbolic link, and then renaming the symbolic link to the desired target name. We can ensure that the target directory name will only appear to the webserver in the volume filesystem once the pull is complete.
+
+## Usage
+
+In the configuration of the git-sync container, set the optional GIT_SYNC_ATOMIC environment variable to true. The GIT_SYNC_SUBDIR environment variable must also be set to a non-empty string (otherwise an error will result). This subdirectory will be created within the shared volume's root directory, specified by GIT_SYNC_DEST (which defaults to "/git" if unspecified). If specified, GIT_SYNC_DEST should match the mountPath for git-sync's mounted emptyDir volume.
```

Comment:
The Kube folks may have their own opinions, but letting people set `GIT_SYNC_SUBDIR` (as opposed to just picking something for them) seems like it adds complexity but not a lot of value. Putting the Git repo in one dir if atomic is off and a subdir of that if atomic is on may also be a bit for users to remember.